mllam_data_prep/ops/cropping.py

```diff
@@ -0,0 +1,201 @@
+
+    # latlon to (x, y, z) on unit sphere
+    da_ref_xyz = _latlon_to_unit_sphere_xyz(da_lat=da_lat_ref, da_lon=da_lon_ref)
+
+    chull_lam = SphericalPolygon.convex_hull(da_ref_xyz.values)
+
+    # call .load() to avoid using dask arrays in the following apply_ufunc
+    da_interior_mask = xr.apply_ufunc(
+        chull_lam.contains_lonlat, da_lon.load(), da_lat.load(), vectorize=True
+    ).astype(bool)
+    da_interior_mask.attrs[
+        "long_name"
+    ] = "contained in convex hull of source dataset (da_ref)"
+
+    return da_interior_mask
+
+
+def _latlon_to_unit_sphere_xyz(
+    da_lat: xr.DataArray, da_lon: xr.DataArray
+) -> xr.DataArray:
+    """
+    Convert lat/lon coordinates to (x, y, z) on the unit sphere.
+
+    Parameters
+    ----------
+    da_lat : xarray.DataArray
+        Latitude coordinates.
+    da_lon : xarray.DataArray
+        Longitude coordinates.
+
+    Returns
+    -------
+    xr.DataArray
+        The (x, y, z) coordinates on the unit sphere as an xarray.DataArray
+        with dimensions (grid_index, component).
+    """
+    pts_xyz = np.array(sg.vector.lonlat_to_vector(da_lon, da_lat)).T
+    da_xyz = xr.DataArray(
+        pts_xyz, coords=da_lat.coords, dims=list(da_lat.dims) + ["xyz"]
+    )
+    return da_xyz
+
+
+def distance_to_convex_hull_boundary(
+    ds: xr.Dataset, ds_reference: xr.Dataset, grid_index_dim: str = "grid_index"
+) -> xr.DataArray:
+    """
+    For all points in `ds` that are external to the convex hull of the points in
+    `ds_reference`, calculate the minimum distance to the convex hull boundary.
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        The dataset for which to calculate the distance.
+    ds_reference : xarray.Dataset
+        The reference dataset from which to calculate the convex hull boundary.
+    grid_index_dim : str, optional
+        The name of the grid index dimension in `ds` and `ds_reference`.
+
+    Returns
+    -------
+    xarray.DataArray
+        The minimum distance to the convex hull boundary.
+    """
+    # rename the grid index dimension in ds_reference to avoid conflicts (since
+    # the grid index dimension otherwise has the same name in both datasets,
+    # and later we will want to find the minimum distance for each point in ds
+    # to all points in ds_reference)
+    ds_reference_separate_gridindex = ds_reference.rename(
+        {grid_index_dim: "grid_index_ref"}
+    )
+
+    # create a mask from the convex hull of ds_reference for the grid points in ds
+    da_ch_mask = create_convex_hull_mask(
+        ds=ds, ds_reference=ds_reference_separate_gridindex
+    )
+
+    # only consider points that are external to the convex hull
+    ds_exterior = ds.where(~da_ch_mask, drop=True)
+
+    da_xyz = _latlon_to_unit_sphere_xyz(ds_exterior.lat, ds_exterior.lon)
+    da_xyz_ref = _latlon_to_unit_sphere_xyz(
+        ds_reference_separate_gridindex.lat, ds_reference_separate_gridindex.lon
+    )
```

Comment:
Here `.lat` and `.lon` is explicitly used, putting a requirement on these exact names being present in the dataset. This is strange as in `create_convex_hull_mask` the function `_get_latlon_coords` is used to not make this assumption. This also causes an error for the same reason as above, that these coordinates will already be renamed to x/y at this point.

Reply:
yes, these direct referencing of `.lat` and `.lot` here was a mistake. I have removed this in commit https://github.com/mllam/mllam-data-prep/pull/45/commits/ba467183ec88959aa529fc0574daf1c82b5d1a1c